radicle-node/src/bounded.rs

```diff
@@ -18,6 +18,34 @@ impl<T, const N: usize> BoundedVec<T, N> {
         BoundedVec { v: Vec::new() }
     }
 
+    /// Build a `BoundedVec` by consuming from the given iterator up to its limit.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use radicle_node::bounded;
+    ///
+    /// let mut iter = (0..4).into_iter();
+    /// let bounded: bounded::BoundedVec<i32,3> = bounded::BoundedVec::collect_from(&mut iter);
+    ///
+    /// assert_eq!(bounded.len(), 3);
+    /// assert_eq!(iter.count(), 1);
+    /// ```
+    pub fn collect_from<I: Iterator<Item = T>>(iter: &mut I) -> Self {
```

Comment:
A couple of things:
1. This is the signature of `FromIterator::from_iter` so you can implement that and we can use `.collect()`
2. Perhaps it makes more sense to use:
```rust
BoundedVec { v: iter().into_iter().tak(Self::max()).collect() }
```
3. If we have a standalone function, i.e. not `from_iter`, perhaps we could return the rest of the iterator so that the caller can decide what to do with the rest of the iterator.

Reply:
"This is the signature of FromIterator::from_iter so you can implement that and we can use .collect()"

Should collect() require collecting all items into the new collection?  I was unsure because of this.  One thing I miss about haskell is clearly defined rules and expectation of type classes.

Reply:
Oh wait, I'm looking at this again and realise you're taking an `&mut I`, which isn't the same as `FromIterator`! Sorry, was looking at this last night wait for a flight home :sweat_smile: This makes sense and looks good to me :+1:

Reply:
Can this function not take an `I` though?

Reply:
> Can this function not take an owned `I` though?

Would this not make it more restrictive?

Reply:
I think it makes more sense to take `&mut` because, as @slack-coder pointed out to me, you're left with the unconsumed portion of the iterator. This is a good thing from my perspective. Say if we did do multiple ref announcements that are windowed by this `236` limit, then you would want to repeat the `BoundedVec` construction until the iterator is consumed.

Reply:
Oh, gotcha :+1: